### ISSUE TITLE
Maximum number of band 

### ISSUE THREAD

**Issue by lolob** [remicres/otbtf#38]
Hi,
I have 10 bands of image (set A) and 12 bands of image (set B). I run BandMathX to create two separate tif files: setA.tif and setB.tif.  Patch sampling was created for both (16x16). Then, I created two model based on simple_fcn.py; one for the 10 bands, and the other one for 12 bands. I make sure that the X placeholders were changed based on the number of bands. Then, I run the TensorflowModelTrain* on both model (defaults parameters) to see some stats on the screen. The SetA shown some numbers on kappa and overall accuracy. However, the SetB just shown kappa=0 and overall accuracy=0.5 even after 100 epoch. By the way, I just want to see result of two classes only. The question, is OTBTF has limitation on number of band that can be used? 

Thanks.
*mdl4eo/otbtf2.0:gpu

**Comment by remicres** [comment]
Hi @lolob ,
There is no limitation on band size, image size, image number, and source number. Anything is possible ;)
My experience is, when a deep net does not converge well, it is often due to the data.

- Are my data well normalized ? (the model can take care of that internally, but at some point you must have [-1, +1] or [0, 1] values that reach the first trainable layers
- Are my patches corrupted? I mean, sometimes to extract patches with nodata values. I see you use otbtf2.0: there is an useful iption in `PatchesExtraction`, to reject patches having at least one nodata value (you can change the nodata value for each source). I recommend to use that: this way, you are 100% sure that your patches are clean :)
- Do I have enough samples? Deep learning models sometimes requires a lot of data. I guess a few thousands patches are a start. The bigger the model is, the more data you might need to train this model from scratch. Of course you can write some python code to do transfer learning, to train partially a model (only a few layers) from less samples. But in this case, you must have a pre-trained model to do that. Anyway I would say that it's risky to train models on small amount of samples!
- Play with learning rate, batch size. Sometimes, a small batch size helps (sometimes not!), just do not hesitate to play with those parameters. Maybe the default ones are not quite suited to your dataset?

Hope that helps!

**Comment by lolob** [comment]
Thanks. By the way, I purchased your book and I found very useful.
